Client/Frontend/Browser/BrowserPrompts.swift

```diff
@@ -35,12 +42,33 @@ class JSPromptAlertController: UIAlertController {
 protocol JSAlertInfo {
     func alertController() -> JSPromptAlertController
     func cancel()
+    /// Informs JSPromptAlertController that the completion handler has already been called.
+    var didComplete: Bool { get }
 }
 
-struct MessageAlert: JSAlertInfo {
+class MessageAlert: JSAlertInfo {
     let message: String
     let frame: WKFrameInfo
-    let completionHandler: () -> Void
+    /// Do not access this directly. Use completionHandler()
+    private var _completionHandler: (() -> Void)?
+    
+    init(message: String, frame: WKFrameInfo, completionHandler: @escaping () -> Void) {
+        self.message = message
+        self.frame = frame
+        self._completionHandler = completionHandler
+    }
+    
+    var didComplete: Bool {
+        _completionHandler == nil
```

Comment:
For clarity as you skim over code, we shy away from not using keywords, if optional. So this would need a `return`.

Also applies in the other places this appears in these changes.

Reply:
Fixed